mlton/front-end/front-end.fun

```diff
@@ -40,6 +40,30 @@ fun lexAndParse (source: Source.t, ins: In.t): Ast.Program.t =
                Ast.Program.T []
             end
       val () = Ast.Program.checkSyntax result
+
+      (* Outputs AST to a file if Control.keepAST is true *)
+      val () =
+        if !Control.keepAST
+          then
+            let
+               val inputFile = File.toString (!Control.inputFile)
```

Comment:
The `FILE` signature has `type t = string`, so it is really just a collection of convenience functions for manipulating file names (as strings), rather than an abstract datatype representing file names.

Using something like:
```SML
File.withAppend(concat [!Control.inputFile, ".ast"], fn outputStream => ...)
```
might be the most concise.